addons/crm/models/crm_lead.py

```diff
@@ -1126,6 +1144,48 @@ def convert_opportunity(self, partner_id, user_ids=False, team_id=False):
 
         return True
 
+    def _get_potential_lead_duplicates_domain(self, lead, include_lost=False):
+        """ Returns a domain that can be used to retrieve the potential lead duplicates.
+            see: `_get_potential_lead_duplicates`
+            :param lead: Reference lead
+            :param boolean include_lost: If true, the leads considered as lost will be included in the results
+        """
+        domain = []
+        for normalized_email in [tools.email_normalize(email) for email in tools.email_split(lead.email_from)]:
+            if isinstance(normalized_email, str) and '@' in normalized_email:
+                _, _, email_domain = normalized_email.rpartition('@')
+                if is_common_domain(email_domain):
+                    domain.append(('email_normalized', '=', normalized_email))
+                else:
+                    domain.append(('email_normalized', '=ilike', '%@' + email_domain))
+
+        if lead.partner_id:
+            domain.append(('partner_id', '=', lead.partner_id.id))
+
+        if len(domain) == 0:
+            return False
+        if len(domain) > 1:
+            domain = ['|'] * (len(domain) - 1) + domain
```

Comment:
```suggestion
        if not domain:
            return False
        domain = expression.OR([[d] for d in domain])
```
Inner list literal would be unnecessary if `domain` was already a list of lists of tuple.